lib/Parser/Parser.yy

```diff
@@ -385,37 +442,61 @@ function_template_arguments:
 
 var_stmt:
       varKw ident type_opt initializer_opt semi
-    { std::cerr << "Parsed var statement" << std::endl; }
+    {
+        $$ = ctx.getASTMemoryArena()
+          .create<VarDecl>(sm.getSourceLocFromToken($2), $2.getLexeme().str(), $3, $4);
+        std::cerr << "Parsed var declaration: " << $2.getLexeme().str() << std::endl; 
+    }
     ;
 
 type_opt:
       /* empty */
+         { $$ = nullptr; }
     | colon type
+         { $$ = $2; }
     ;
 
 initializer_opt:
       /* empty */
+         { $$ = nullptr; }
     | equal expression
+         {
+            if ($2 == nullptr) {
+                error("Missing expression after '='");
+                YYERROR;
```

Comment:
no reason to trigger the parser's error recovery mechanism here? also this isn't supposed to happen when the whole AST is implemented